### PR TITLE
Adding some comments to P2P classes

### PR DESCRIPTION
**Pull request by vncoelho**
@Tommo-L @shargon @erikzhang @yongjiema @igormcoelho,
we need more comments to some of our classes.

They will surely help us and others. Sometimes it is hard to comprehend, even after you understanding the code. When you stay away from it for some time you forget some important parts.

I am planning an effort for commenting some of the P2P classes, as well as the consensus service.

Thus, if you have any additional idea for the `Peers` and `LocalNode` file, fell free to comment it here.

The idea is to leave this PR open for, at least, 10-15 days, sometime that we can all include some comments we want.

### CODE REVIEW

**Comment by shargon** [comment]
Could me moved to https://github.com/neo-project/neo/pull/1185 ?

**Comment by vncoelho** [comment]
No, @shargon, it is different things.

**Comment by vncoelho** [comment]
@shargon, adding to my previous comment since I saw that you got confused.
The other PR is a renaming, which, essentially, is different than commenting the existing code.

File: neo/Network/P2P/Peer.cs [hunk 1]
```diff
@@ -62,10 +62,16 @@ static Peer()
             localAddresses.UnionWith(NetworkInterface.GetAllNetworkInterfaces().SelectMany(p => p.GetIPProperties().UnicastAddresses).Select(p => p.Address.Unmap()));
         }
 
+        /// <summary>
+        /// Tries to add a set ff peers to an immutable hashset of UnconnectedPeers
```
Comment (lock9) [inline neo/Network/P2P/Peer.cs]: What is an "ff peer"?

Comment (shargon) [reply neo/Network/P2P/Peer.cs]: it's of i think

File: neo/Helper.cs [hunk 2]
```diff
@@ -248,6 +248,9 @@ internal static IPAddress Unmap(this IPAddress address)
             return address;
         }
 
+        /// <summary>
+        /// Checks IPv4 and IPv6 formats and unmap, if necessary
+        /// </summary>
         internal static IPEndPoint Unmap(this IPEndPoint endPoint)
```
Comment (shargon) [inline neo/Helper.cs]: Could you add the info for the parameters too?

File: neo/Network/P2P/LocalNode.cs [hunk 3]
```diff
@@ -122,6 +122,11 @@ public IEnumerable<IPEndPoint> GetUnconnectedPeers()
             return UnconnectedPeers;
         }
 
+        /// <summary>
+        /// Override of abstract class that is triggered when UnconnectedPeers is empty
+        /// Performs a BroadcastMessage with the command `MessageCommand.GetAddr`, which, eventually, tells all known connections
+        /// If there are no connected peers it will try with the default, respecting MaxCountFromSeedList limit
+        /// </summary>
         protected override void NeedMorePeers(int count)
```
Comment (shargon) [inline neo/Network/P2P/LocalNode.cs]: Count parameter information too

**Comment by vncoelho** [comment]
@shargon, I still plan to add more coments. Let is open a little bit more.

**Comment by shargon** [comment]
@vncoelho cool, tell me when you finish!

**Comment by ShawnYun** [comment]
I have writen many Chinese comments of P2P codes, maybe I can writen a English version of comments. I think it is import for us to read codes.

**Comment by vncoelho** [comment]
@ShawnYun, very good!
When can you post them here?

If you want you can open a PR for changing this branch and we can revise it before merging.

**Comment by ShawnYun** [comment]
@vncoelho I will post it as soon as possible.

File: src/neo/Network/P2P/Peer.cs [hunk 4]
```diff
@@ -141,7 +167,8 @@ private void OnStart(ChannelsConfig config)
             MaxConnections = config.MaxConnections;
             MaxConnectionsPerAddress = config.MaxConnectionsPerAddress;
 
-            timer = Context.System.Scheduler.ScheduleTellRepeatedlyCancelable(0, 5000, Context.Self, new Timer(), ActorRefs.NoSender);
+            // schedule time to trigger `OnTimer` event every TimerMillisecondsInterval ms
+            timer = Context.System.Scheduler.ScheduleTellRepeatedlyCancelable(0, TimerMillisecondsInterval, Context.Self, new Timer(), ActorRefs.NoSender);
```
Comment (erikzhang) [inline src/neo/Network/P2P/Peer.cs]: I think such changes actually reduce readability.

Comment (vncoelho) [reply src/neo/Network/P2P/Peer.cs]: Let's revert this one. It, in fact, should had not come here.
It was from a previous PR and was merged for some reason.

**Comment by vncoelho** [comment]
@erikzhang, what is your opnion about these comments?
Which ones do you think are relevant?

After this PR, the plan is to document consensus. However, consensus will be easier.

**Pull request merged**